services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/bsm/OdeBsmJsonRepositoryImpl.java

```diff
@@ -0,0 +1,136 @@
+package us.dot.its.jpo.ode.api.accessors.bsm;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import us.dot.its.jpo.geojsonconverter.DateJsonMapper;
+import us.dot.its.jpo.ode.api.ConflictMonitorApiProperties;
+import us.dot.its.jpo.ode.model.OdeBsmData;
+
+@Component
+public class OdeBsmJsonRepositoryImpl  implements OdeBsmJsonRepository{
+
+    @Autowired
+    private MongoTemplate mongoTemplate;
+
+    @Autowired
+    ConflictMonitorApiProperties props;
+
+    private ObjectMapper mapper = DateJsonMapper.getInstance();
+
+    private String collectionName = "OdeBsmJson";
+
+    public static Double[] calculateLatitudes(Double centerLatitude, Double radiusInMeters) {
+        Double latDiff = radiusInMeters / 111319.9; // Approximate degrees latitude per meter
```

Comment:
issue: We reuse this "111319.9" value in a couple places. Would it make sense to pull out to a class level variable for reuse and readability? Also, this description is a bit misleading - I don't think there is 111319.9 degrees latitude in a single meter, but rather the other way around. Though I'm not sure where we're getting this value from? Quick search on meters per degree latitude shows several sources with 111,111.1 meters per degree latitude (though I've seen 111,139 in a few places as well, maybe we were meaning to use that?).

Reply:
I refactored this using org.geotools